lib/pf/Authentication/Source/GoogleSource.pm

```diff
@@ -45,6 +46,21 @@ sub lookup_from_provider_info {
     person_modify( $pid, firstname => $info->{given_name}, lastname => $info->{family_name}, email => $info->{email}, gender => $info->{gender} );
 }
 
+
+=head2 additional_client_attributes
+
+Supply a google-apps domain filter (hd parameter). See below
+https://developers.google.com/identity/protocols/OpenIDConnect#hd-param
+
+=cut
+
+sub additional_client_attributes {
```

Comment:
This subroutine doesn't seem to be used anywhere, is this because I'm missing some of the magic from Net::OAuth2 ?

Reply:
Its part of the OAuthSource.pm and its used by the linkedin Oauth module to pass an additional parameter needed by it. I copied that same model to pass the HD attribute to google.